File: roteiro4/experimento1/main.py
import time

N = 10_000
buf = bytearray(N)

t0 = time.ticks_us()
for i in range(N):
    buf[i] = i & 0xFF   # escrita em RAM (heap)
t1 = time.ticks_us()

dt_ms = time.ticks_diff(t1, t0) / 1000
print("RAM write ({} bytes): {:.3f} ms".format(N, dt_ms))

# RAM write (10000 bytes): 153.715 ms
# RAM write (10000 bytes): 153.715 ms
# RAM write (10000 bytes): 153.720 ms
# 153.716

File: roteiro4/experimento1/main.c
#include <stdio.h>
#include "pico/stdlib.h"

int main() {
    stdio_init_all();
    const int N = 10000;
    static uint8_t buf[10000];

    sleep_ms(5000);

    printf("Hello, World!\n");

    absolute_time_t t0 = get_absolute_time();
    for (int i = 0; i < N; i++) buf[i] = i & 0xFF;
    absolute_time_t t1 = get_absolute_time();

    int64_t us = absolute_time_diff_us(t0, t1);
    printf("RAM write (%d bytes): %.3f ms\n", N, us/1000.0);

    while (1) tight_loop_contents();
}
// RAM write (10000 bytes): 0.001 ms
// RAM write (10000 bytes): 0.001 ms
// RAM write (10000 bytes): 0.001 ms